src/operation/select-rule.ts

```
@@ -237,16 +237,29 @@ export const selectOptimalRule = (
+    },
   );
 
-  if (fullSelectableRules.length > 0) {
+  // Try full selection first
+  if (partitionedRules.fullSelectable.length > 0) {
     // Select first rule from sorted list (already the best)
-    // Safe: fullSelectableRules.length > 0 guaranteed by preceding check
+    // Safe: fullSelectable.length > 0 guaranteed by preceding check
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    const bestRule = fullSelectableRules[0]!;
+    const bestRule = partitionedRules.fullSelectable[0]!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)